app/src/main/java/org/sopt/and/ui/signup/SignUpScreen.kt

```diff
@@ -42,37 +43,53 @@ import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.input.ImeAction
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.lifecycle.compose.LocalLifecycleOwner
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.flowWithLifecycle
 import androidx.lifecycle.viewmodel.compose.viewModel
 import org.sopt.and.R
-import org.sopt.and.ui.component.textfield.SignUpTextField
+import org.sopt.and.ui.ShowSnackBar
+import org.sopt.and.ui.component.textfield.UserInfoTextField
 import org.sopt.and.ui.signup.SignUpViewModel.Companion.EXTRA_SIGNUP_IMAGE_LIST
+import org.sopt.and.ui.toast
 
 
 @Composable
 fun SignUpScreen(
-    navigateToLogIn: (id: String, password: String) -> Unit,
+    navigateToLogIn: (signUpState:SignUpState) -> Unit,
     navigateToBack: () -> Unit
 ) {
     val viewModel = viewModel<SignUpViewModel>()
     val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val coroutineScope = rememberCoroutineScope()
 
     var isPasswordVisible by remember { mutableStateOf(false) }
 
-    val signUpState = viewModel.signupState.collectAsStateWithLifecycle()
-    val id = signUpState.value.email
-    val password = signUpState.value.password
+    val signUpState by viewModel.signupState.collectAsStateWithLifecycle()
 
     val snackbarHostState = remember { SnackbarHostState() }
-    val snackbarMessage by viewModel.snackbarMessage
 
-    if (snackbarMessage.isNotEmpty()) {
-        LaunchedEffect(snackbarHostState) {
-            snackbarHostState.showSnackbar(snackbarMessage)
-            viewModel.clearSnackbarMessage()
-        }
+    LaunchedEffect(viewModel.signUpSideEffect, snackbarHostState, lifecycleOwner) {
+        viewModel.signUpSideEffect.flowWithLifecycle(lifecycle = lifecycleOwner.lifecycle)
+            .collect { sideEffect ->
+                when (sideEffect) {
+                    is SignUpSideEffect.ShowToast -> {
+                        context.toast(sideEffect.message)
+                    }
+
+                    is SignUpSideEffect.ShowSnackBar -> {
+                        snackbarHostState.ShowSnackBar(
+                            coroutineScope,
+                            sideEffect.message,
+                            context
+                        )
+                    }
+                }
+            }
```

Comment:
```suggestion
    LaunchedEffect(Unit) {
        viewModel.signUpSideEffect.flowWithLifecycle(lifecycle = lifecycleOwner.lifecycle)
            .collect { sideEffect ->
                when (sideEffect) {
                    is SignUpSideEffect.ShowToast -> {
                        context.toast(sideEffect.message)
                    }

                    is SignUpSideEffect.ShowSnackBar -> {
                        snackbarHostState.ShowSnackBar(
                            coroutineScope,
                            sideEffect.message,
                            context
                        )
                    }
                }
            }
```

Q) 위와 같이 변경되면 앱 동작에는 어떤 변화가 일어날까요?

Reply:
key값이 사라지기 때문에 처음 함수가 composition될때 한번 이외에는 실행되지않습니다.Toast나 SnackBar의 경우  recompose안되는경우도 있기때문에 key값은 필요하다고 생각합니다